### PR TITLE
[DOCS] How to connect to your data in BigQuery

### PR DESCRIPTION
**Pull request by Shinnnyshinshin**
## Changes proposed in this pull request:
- How-to-guide for BigQuery
- Associated test scripts for `python` and `yaml` configuration.
- Removed error that was preventing GE from creating bigquery tables without specifying `batch_spec_passthrough`. Removing this error allows for sqlalchemy_batch_data.py to behave in the same way as CLI. Please see `great_expectations/cli/batch_request.py` line 268-274 as a reference. 

## Running the tests
- Currently Azure does not run the integration tests for `BigQuery`, but they can be run locally. 
- In order to run the tests locally:

1. Complete GCS authentication locally, which involves setting `GOOGLE_APPLICATION_CREDENTIALS` environment variable.
2. uncomment the tests in the `test_script_runner.py` file (for `bigquery_yaml_example.py` and `bigquery_python_example.py`)
3. Run pytest by using the following command`pytest -v --docs-tests -m docs --no-spark tests/integration/test_script_runner.py`

## Checklist
- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

File: tests/integration/test_script_runner.py [hunk 1]
```diff
@@ -23,6 +24,20 @@ class BackendDependencies(enum.Enum):
...
 
 docs_test_matrix = [
+    # {
+    #     "user_flow_script": "tests/integration/docusaurus/connecting_to_your_data/database/bigquery_yaml_example.py",
+    #     "data_context_dir": "tests/integration/fixtures/no_datasources/great_expectations",
+    #     "data_dir": "tests/test_sets/taxi_yellow_trip_data_samples",
+    #     "util_script": "tests/integration/docusaurus/connecting_to_your_data/database/util.py",
+    #     "extra_backend_dependencies": BackendDependencies.BIGQUERY,
+    # },
+    # {
```
Comment (Shinnnyshinshin) [inline tests/integration/test_script_runner.py:34]: These will be re-enabled once Azure pipelines are connected with GGS

File: great_expectations/execution_engine/sqlalchemy_batch_data.py [hunk 2]
```diff
@@ -139,11 +139,6 @@ def __init__(
                 # mssql expects all temporary table names to have a prefix '#'
                 if engine.dialect.name.lower() == "mssql":
                     generated_table_name = f"#{generated_table_name}"
-                if engine.dialect.name.lower() == "bigquery":
-                    raise ValueError(
-                        "No BigQuery dataset specified.  Include bigquery_temp_table in "
-                        "batch_spec_passthrough or a specify a default dataset in engine url"
-                    )
```
Comment (Shinnnyshinshin) [inline great_expectations/execution_engine/sqlalchemy_batch_data.py:146]: This error was preventing GE from creating bigquery tables without specifying `batch_spec_passthrough`.  Removing this error allows for `sqlalchemy_batch_data.py` to behave in the same way as CLI

Please see `great_expectations/cli/batch_request` line 268-274

File: tests/integration/test_script_runner.py (hunk 1 above)
Comment (Shinnnyshinshin) [reply tests/integration/test_script_runner.py:34]: in order to run the tests locally, un-comment these lines

File: docs/guides/connecting_to_your_data/database/bigquery.md [hunk 3]
```diff
@@ -1,4 +1,179 @@
...
+```
+
+:::note
+Currently BigQuery does not allow for the creation of temporary tables as the result of a query.  Therefore, Great Expectations will automatically create a permanent table with a random UUID in the `GCP_PROJECT` and `BIGQUERY_DATASET` defined in the connection string.
+Great Expectations also allows for a named table to be used as a temporary table, and the name can be passed in as a `batch_spec_passthrough` parameter. In the following example we are using a table named `ge_temp`.
+:::  
+
+```python file=../../../../tests/integration/docusaurus/connecting_to_your_data/database/bigquery_yaml_example.py#L61-L76
+```
+
```
Comment (Shinnnyshinshin) [inline docs/guides/connecting_to_your_data/database/bigquery.md]: This is the section where both "versions" of the `batch_spec` (with and without the `batch_spec_passthrough`) are described.  Requesting feedback on how this part looks and flows

File: tests/integration/docusaurus/connecting_to_your_data/database/bigquery_python_example.py [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+context.add_datasource(**datasource_config)
+
+# First test for RuntimeBatchRequest using a query. bigquery_temp_table name is generated automatically
+batch_request = RuntimeBatchRequest(
+    datasource_name="my_bigquery_datasource",
+    data_connector_name="default_runtime_data_connector_name",
+    data_asset_name="default_name",  # this can be anything that identifies this data
+    runtime_parameters={"query": "SELECT * from demo.taxi_data LIMIT 10"},
+    batch_identifiers={"default_identifier_name": "something_something"},
+)
```
Comment (Aylr) [inline tests/integration/docusaurus/connecting_to_your_data/database/bigquery_python_example.py]: Please remove this in favor of a ticket in JIRA.

File: tests/integration/docusaurus/connecting_to_your_data/database/bigquery_python_example.py [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+# NOTE: The following code is only for testing and can be ignored by users.
+assert isinstance(validator, ge.validator.validator.Validator)
+
+# First test for BatchRequest naming a table, bigquery_temp_table name is generated automatically
+batch_request = BatchRequest(
+    datasource_name="my_bigquery_datasource",
+    data_connector_name="default_inferred_data_connector_name",
+    data_asset_name="taxi_data",  # this is the name of the table you want to retrieve
+)
```
Comment (Aylr) [inline tests/integration/docusaurus/connecting_to_your_data/database/bigquery_python_example.py]: Please remove this in favor of a ticket in JIRA.

File: tests/integration/docusaurus/connecting_to_your_data/database/bigquery_yaml_example.py [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+context.add_datasource(**yaml.load(datasource_yaml))
+
+# First test for RuntimeBatchRequest using a query. bigquery_temp_table name is generated automatically
+batch_request = RuntimeBatchRequest(
+    datasource_name="my_bigquery_datasource",
+    data_connector_name="default_runtime_data_connector_name",
+    data_asset_name="default_name",  # this can be anything that identifies this data
+    runtime_parameters={"query": "SELECT * from demo.taxi_data LIMIT 10"},
+    batch_identifiers={"default_identifier_name": "something_something"},
+)
```
Comment (Aylr) [inline tests/integration/docusaurus/connecting_to_your_data/database/bigquery_yaml_example.py]: Please remove this in favor of a ticket in JIRA.

File: tests/integration/docusaurus/connecting_to_your_data/database/bigquery_yaml_example.py [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+
+# NOTE: The following code is only for testing and can be ignored by users.
+assert isinstance(validator, ge.validator.validator.Validator)
+
+# First test for BatchRequest naming a table, bigquery_temp_table name is generated automatically
+batch_request = BatchRequest(
+    datasource_name="my_bigquery_datasource",
+    data_connector_name="default_inferred_data_connector_name",
+    data_asset_name="taxi_data",  # this is the name of the table you want to retrieve
+)
```
Comment (Aylr) [inline tests/integration/docusaurus/connecting_to_your_data/database/bigquery_yaml_example.py]: Please remove this in favor of a ticket in JIRA.

**Review by Aylr — CHANGES_REQUESTED** [review]
Change request as we discussed.

File: docs/guides/connecting_to_your_data/database/bigquery.md [hunk 8]
```diff
@@ -1,4 +1,179 @@
...
+Here is an example of loading data by specifying a SQL query.
+
+```python file=../../../../tests/integration/docusaurus/connecting_to_your_data/database/bigquery_yaml_example.py#L52-L58
+```
+
+:::note
+Currently BigQuery does not allow for the creation of temporary tables as the result of a query.  Therefore, Great Expectations will automatically create a permanent table with a random UUID in the `GCP_PROJECT` and `BIGQUERY_DATASET` defined in the connection string.
+Great Expectations also allows for a named table to be used as a temporary table, and the name can be passed in as a `batch_spec_passthrough` parameter. In the following example we are using a table named `ge_temp`.
+:::
```
Comment (Aylr) [inline docs/guides/connecting_to_your_data/database/bigquery.md]: Also take this note out as discussed.

**Comment by Shinnnyshinshin** [comment]
Hi @Aylr thanks for the review. I wanted to check what you thought of a simplified note to the user.  I feel like there would be a benefit to explaining why this parameter exists and how it should be used.  Here is a "simplified" version. 

Screen Shot 2021-06-18 at 11 29 34 AM

**Review by Aylr — APPROVED** [review]
LGTM

**Pull request merged**